examples/resolvers-planetscale/grafbase/resolvers/products/update.ts

```diff
@@ -0,0 +1,94 @@
+import { connect, cast } from "@planetscale/database";
+
+const config = {
+  host: process.env.DATABASE_HOST,
+  username: process.env.DATABASE_USERNAME,
+  password: process.env.DATABASE_PASSWORD,
+};
+
+const conn = connect(config);
+
+export const options = {
+  cast(field, value) {
+    switch (field.name) {
+      case "id": {
+        return String(value);
+      }
+      case "onSale": {
+        return Boolean(value);
+      }
+      default: {
+        return cast(field, value);
+      }
+    }
+  },
+};
+
+export default async function ProductsUpdate(_, args) {
+  const {
+    by: { id, slug },
+    input: { name: newName, slug: newSlug, price: newPrice, onSale: newOnSale },
+  } = args;
+
+  let updateStatement = "UPDATE Products SET ";
+  let params = [];
+
+  if (newName !== undefined) {
+    updateStatement += "name = ?, ";
+    params.push(newName);
+  }
+  if (newSlug !== undefined) {
+    updateStatement += "slug = ?, ";
+    params.push(newSlug);
+  }
+  if (newPrice !== undefined) {
+    updateStatement += "price = ?, ";
+    params.push(newPrice);
+  }
+  if (newOnSale !== undefined) {
+    updateStatement += "onSale = ?, ";
+    params.push(newOnSale);
+  }
+
+  if (params.length === 0) {
+    throw new Error("At least one field to update must be provided.");
+  }
+
+  updateStatement = updateStatement.slice(0, -2);
+
+  if (id !== undefined) {
+    updateStatement += " WHERE id = ?";
+    params.push(id);
+  } else if (slug !== undefined) {
+    updateStatement += " WHERE slug = ?";
+    params.push(slug);
+  } else {
+    // Throw new GraphQLError('ID or Slug must be provided')
+  }
+
+  let selectStatement;
+  let selectParams;
```

Comment:
Just in case, I'm not familiar with PlanetScale, but with `UPDATE` you can usually add a `RETURNING *` which returns all updated data.

Reply:
I didn't think `RETURNING *` was supported by MySQL. Is that a new thing?

Reply:
might only be in PostgreSQL 🤔

Reply:
Nice find, I originally started with using procedures for all operations in fact but thought it might be a tad over the top for an example 😆